### ISSUE TITLE
chinese char support

### ISSUE THREAD

**Issue by matrix3d** [apache/royale-asjs#405]
pls

**Comment by carlosrovira** [comment]
Hi Lizhi, 
can you provide more info? what means "pls"?
Don't know others in the project, but talking about me, I don't know almost anything about Chinese char, so maybe this is a very good opportunity for you to submit a patch to support this. What do you think? it's possible?
Thanks

**Comment by matrix3d** [comment]
trace("中国")

**Comment by greg-dove** [comment]
Hi Lizhi,

That works for me. I can see that trace work in the browser console.

Do you have the source files saved as utf8?

If that doesn't work, maybe another possible thing to check is to try
(windows example):
set JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF-8 -Xmx1024m

Others may have other ideas...

**Comment by matrix3d** [comment]
```
涓夋湀 19, 2019 2:37:15 涓嬪崍 com.google.javascript.jscomp.LoggerErrorManager println
璀﹀憡: F:/proj/spriteflexjs/examples/bin/js-debug/flash/__native/te/CharSet.js:144: WARNING - unreachable code
    if (false) {
               ^^

涓夋湀 19, 2019 2:37:15 涓嬪崍 com.google.javascript.jscomp.LoggerErrorManager println
璀﹀憡: F:/proj/spriteflexjs/examples/bin/js-debug/flash/text/TextField.js:380: WARNING - Variable referenced before declaration: i
    for (i = 0; i < clen; i++) {
         ^

涓夋湀 19, 2019 2:37:15 涓嬪崍 com.google.javascript.jscomp.LoggerErrorManager println
璀﹀憡: F:/proj/spriteflexjs/examples/bin/js-debug/flash/text/TextField.js:380: WARNING - Variable referenced before declaration: i
    for (i = 0; i < clen; i++) {
                ^

涓夋湀 19, 2019 2:37:15 涓嬪崍 com.google.javascript.jscomp.LoggerErrorManager println
璀﹀憡: F:/proj/spriteflexjs/examples/bin/js-debug/flash/text/TextField.js:380: WARNING - Variable referenced before declaration: i
    for (i = 0; i < clen; i++) {
                          ^

涓夋湀 19, 2019 2:37:15 涓嬪崍 com.google.javascript.jscomp.LoggerErrorManager printSummary
璀﹀憡: 0 error(s), 4 warning(s), 98.4% typed
```

**Comment by Harbs** [comment]
@matrix3d Those are Google Closure warnings.

You can ignore them or fix them in your code.

**Comment by carlosrovira** [comment]
Hi Lizhi,
please, try to add more phrasing to your tickets/comments, if you writing is so cryptic or there's almost not explanation is difficult to follow you. We're a community, and I think we all need normal conversations. Maybe foreign language is difficult for you, but the same happens to me. And we all should make a little effort. Hope you understand. Thanks in advance :)

**Comment by yestaro** [comment]
Hi everyone
I use Traditional Chinese. In my experience, it works fine with Apache Royale.

Hi Lizhi, 
if you don't know how to describe your problems in English,  try to describe in Chinese... may be I can help

**Comment by matrix3d** [comment]
@yestaro 嗯，问题不大了。需要加入这行代码 set JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF-8 -Xmx1024m。但是比较尴尬的是，加入这行代码编译通过了。编译的输出又是乱码了。而且偶尔还是会编译错误，这个编译错误是随机出现的，也可以不管了。就这样吧，能编译就行

**Comment by yestaro** [comment]
@matrix3d 這就有點奇怪。我沒加入任何的JAVA_TOOL_OPTIONS，到現在沒發生過錯誤。
我是用vs code開發的，也許我們編譯的工具是不一樣的。vs code編譯所下達的命令式如下
C:\Program Files (x86)\Common Files\Oracle\Java\javapath\java.exe -jar c:\Users\cwyu0\.vscode\extensions\bowlerhatllc.vscode-nextgenas-0.18.0\bin\asconfigc.jar --sdk "c:\Program Files (x86)\Apache Flex\royale-0.9.6\royale-asjs" --debug=false --project d:\workspace\royale\wec-core\asconfig.json

重點可能是這段 bowlerhatllc.vscode-nextgenas-0.18.0\bin\asconfigc.jar

**Comment by JamesPengTW** [comment]
我這邊也是中文亂碼的問題 @@;  我用的編譯工具是Moonshine 
這樣的問題會是編譯工具本身的問題嗎?
![QQ](https://user-images.githubusercontent.com/49034551/55426650-e623e900-55b7-11e9-8237-45dfd888370d.png)

**Comment by JamesPengTW** [comment]
The tool software I use is Moonshine.In my test, I found that Chinese, Japanese, and Korean could not be displayed correctly. It'is garbled. Has anyone else encountered the same problem as me?
![qqq](https://user-images.githubusercontent.com/49034551/55429474-2d14dd00-55be-11e9-9313-080173aa50d4.png)
Has anyone else encountered the same problem as me?

**Comment by JamesPengTW** [comment]
![qqqq](https://user-images.githubusercontent.com/49034551/55429601-77965980-55be-11e9-9a9e-93581236aac4.png)

**Comment by carlosrovira** [comment]
We found similar problems in our real project, but we used unicode for special latin characters.
Do you think this could be solved maybe in Royale-compiler?
We use VSCode for now, so don't think tooling is a problem here. I think this could be something to address in compiler, or maybe is just config in html template we use

**Comment by JamesPengTW** [comment]
@yestaro 你是用VS Code來做編譯的嗎? 是否中文字的輸出沒問題?

**Comment by yestaro** [comment]
@JamesPengTW 我是用 VS Code。像trace("中文") 這類程式是沒問題的。昨天細看了一下編譯輸出，一些警告訊息，確是亂碼。但那只是警告訊息，不影響輸出

It worked fine like trace("中文"), but the warning message of compiler does not support chinese

**Comment by yestaro** [comment]
@JamesPengTW 我剛用Moonshine試了一下，的確不行。但VS Code是可以的。

It does not support chinese when use Moonshine, but OK with VS Code.
I test it on the same environment, JDK 10 / Royale 0.9.4
compile the source file ApacheRoyaleTest.mxml. 

- output js-debug/ApacheRoyaleTest.js from Moonshine encoded in ANSI
- output js-debug/ApacheRoayleTest.js from VSCode encode in UTF8

I don't know why their output js file has different encoded....

**Comment by piotrzarzycki21** [comment]
> @JamesPengTW 我剛用Moonshine試了一下，的確不行。但VS Code是可以的。
> 
> It does not support chinese when use Moonshine, but OK with VS Code.
> I test it on the same environment, JDK 10 / Royale 0.9.4
> compile the source file ApacheRoyaleTest.mxml.
> 
> * output js-debug/ApacheRoyaleTest.js from Moonshine encoded in ANSI
> * output js-debug/ApacheRoayleTest.js from VSCode encode in UTF8
> 
> I don't know why their output js file has different encoded....

I will raise an issue in Moonshine to investigate that stuff. I will try to fix that if not in version 2.1.0 - I should get results in the next one.

**Comment by piotrzarzycki21** [comment]
@yestaro , @JamesPengTW here is the solution for Moonshine. Probably VSCode have this setup at some point default - I'm not sure. To test that stuff I have used MX [Hello World](https://github.com/apache/royale-asjs/tree/develop/examples/mxroyale/HelloWorld) example and newest nightly build [2115](http://apacheroyaleci.westus2.cloudapp.azure.com:8080/job/royale-asjs/2115/) of Royale.

`<mx:Label text="來做編譯的嗎? 是否中文字的輸出沒問題" x="20" y="20" />`

Go to Windows Environment variables and add -JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF-8
![javatoolsoptions](https://user-images.githubusercontent.com/24554795/55529956-2791c280-56a4-11e9-9759-2442e9953914.png)

I have that option setup for a long time now, so I tried build this Hello World without it and I was able to reproduce your problems. Once you setup above parameter in environment variable restart Moonshine.
At the end of the build you should see something like that if option is setup correctly.

```
: The project 'HelloWorld' has been successfully compiled and optimized.
: 79.393026687 seconds
: Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
: Project Build Successfully.
: Launching application HelloWorld.
: Application HelloWorld started.
```

Let me know if it helps.

**Comment by JamesPengTW** [comment]
The problem is solved!! I am sure that the Chinese, Japanese, Korean and other languages ​​can be displayed correctly. @piotrzarzycki21  you are right! thank you a lot.

@matrix3d @yestaro   it's solution for Moonshine.

**Comment by piotrzarzycki21** [comment]
> The problem is solved!! I am sure that the Chinese, Japanese, Korean and other languages ​​can be displayed correctly. @piotrzarzycki21 you are right! thank you a lot.
> 
> @matrix3d @yestaro it's solution for Moonshine.

Happy to help! :) Thanks for using Moonshine!

**Comment by joshtynjala** [comment]
This issue should be fixed in commit apache/royale-compiler@b95cd6f3d47e1d21f1693347dc7bce676c72b682
